products/sle/main.pm

```diff
@@ -199,6 +199,39 @@ if (get_var('DEV_IMAGE')) {
     }
 }
 
+#This setting is used to enable all extensions/modules for upgrading tests, generate corresponding extensions/modules list for all addons, all extensions, all modules cases.
+if (check_var('SCC_REGISTER', 'installation') && get_var('ALL_ADDONS')) {
+    my $version = get_required_var("HDDVERSION");
+    #Below $common* store the extensions/modules list in common.
+    my $common_addons_12SP3     = 'sdk,pcm,tcm,wsm';
+    my $common_modules_12SP3    = 'pcm,tcm,wsm';
+    my $common_extensions_12SP3 = 'sdk';
+    my $common_addons_11SP4     = 'ha,geo,sdk';
+    #Below $external* store external extensions/modules list on each ARCH.
+    my %external_addons_12SP3
+      = ('x86_64', 'ha,geo,we,live,asmm,contm,lgm,hpcm', 'ppc64le', 'ha,live,asmm,contm,lgm', 's390x', 'ha,geo,asmm,contm,lgm', 'aarch64', 'hpcm');
+    my %external_modules_12SP3    = ('x86_64', 'asmm,contm,lgm,hpcm', 'ppc64le', 'asmm,contm,lgm', 's390x', 'asmm,contm,lgm', 'aarch64', 'hpcm');
+    my %external_extensions_12SP3 = ('x86_64', 'ha,geo,we,live',      'ppc64le', 'ha,live',        's390x', 'ha,geo',         'aarch64', '');
+    # ALL_ADDONS=1:all addons 2:all extensions 3:all modules
+    if ($version =~ "12-SP3") {
+        set_var("SCC_ADDONS", "$common_addons_12SP3,$external_addons_12SP3{get_var('ARCH')}") if (check_var('ALL_ADDONS', '1'));
+        if (check_var('ALL_ADDONS', '2')) {
```

Comment:
Personally, I'm totally against encoding logic into some magic numbers. What's about either splitting it into 3 variables ALL_ADDONS, ALL_MODULES and ALL_MODULES_EXTENSIONS, or using  words for the value, like ALL_ADDONS=modules, ALL_ADDONS=extensions, etc. Then it will be easier to understand. On top, do we always want to overwrite current value? I would keep the value we have if it's set to have more flexibility.